---
title: "GWAS-GAPIT"
author: "P2"
date: "`r Sys.Date()`"
output: html_document
---
Genome-Wide Association Study (GWAS) systematically examines the entire genome of a large group of individuals(diverse population) to identify genetic variations, known as single nucleotide polymorphisms (SNPs), that are statistically associated with specific traits, diseases, or conditions. This approach helps researchers uncover genetic factors contributing to complex traits and diseases, though it primarily identifies correlations and not causation, necessitating further investigations to understand the biological mechanisms involved.

Step 1:Intaliing GAPIT

```{r}
source("http://zzlab.net/GAPIT/GAPIT.library.R")
source("http://zzlab.net/GAPIT/gapit_functions.txt")
```


2. Importing Genotyping data
-GAPIT accepts multiple input dataformats, including both numeric, hapmap, and PLINK genotype formats
-Import the data providing the full path
-Make sure head is set FALSE

```{r}
genotype <- read.table("C:/Users/Arbin/Downloads/Watkins_axiom35k_genotype_for_Tassel.hmp.txt", head = FALSE)

```

3. Import Phenotyping data

-Make sure the header is set to TRUE
```{r}
phenotype <- read.table("C:/Users/Arbin/Downloads/Watkins_phenotype_H2006_for_Tassel.txt", header = TRUE)
```

4. Running GWAS
- Y argument takes phenotypic data
- G argument takes genotypic data
- model stands for which model you want to take into consideration:
  The Mixed Linear Model (MLM) is one of the most effective methods for controlling false positives in GWAS. This model 
  simultaneously incorporates both population structure and cryptic relationship
- SNP.MAF of 0.05 will retain only those SNPs where the less common allele is present in the population at a frequency of 0.05 or higher

```{r}
gwas<-GAPIT(Y=phenotype,G=genotype,PCA.total=5,
  SNP.MAF =0.05,
  model= "MLM",
  Random.model=FALSE
)
  
```
5. Output are in .csv files,PDFs and interactive plots in HTMLs
- Manhattan plots visualize GWAS results by plotting genomic SNP positions on the X-axis and the negative logarithm of P- 
  values from the F-test for assessing SNP-trait associations on the Y-axis. Prominent peaks in the plot indicate 
  significant genomic regions linked to the trait
- The QQ-plot in GWAS assesses the model's ability to account for population structure and familial relatedness by plotting 
  the negative logarithms of P-values against their   expected values under the null hypothesis, with most points expected 
  to lie on the diagonal line, while deviations from this line, primarily in the upper right section,suggest potential 
  trait associations and spurious associations related to insufficient   model correction for such factors
- Association Table
  The GWAS result table provides a detailed summary of appropriate GWAS results. The rows display theresults for each 
  SNP above the user-specified minor allele frequency    threshold. The SNPs sorted by their
  position in the chromosome. The H&B P.Value is FDR adjusted P values by H&B method. 



References:
- Wang, J. & Zhang, Z. GAPIT Version 3: Boosting Power and Accuracy for Genomic Association and Prediction.Genomics. 
     Proteomics Bioinformatics 19, 1–12 (2021).
- Zhang, Z. et al. Mixed linear model approach adapted for genome-wide association studies. Nat. Genet. 42, 355–60(2010).
